### PR TITLE
feat!(deps): bump electron to 29 COMPASS-7868

### PR DESCRIPTION
**Pull request by addaleax**
### Checklist
- [ ] New tests and/or benchmarks are included
- [ ] Documentation is changed or added
- [ ] I have signed the MongoDB Contributor License Agreement (https://www.mongodb.com/legal/contributor-agreement)

## Motivation and Context

- [x] Bugfix
- [ ] New feature
- [x] Dependency update
- [ ] Misc

## Types of changes

- [ ] *Backport Needed*
- [x] Patch (non-breaking change which fixes an issue)
- [ ] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/compass-sidebar/package.json [hunk 1]
```diff
@@ -70,7 +70,7 @@
     "redux-thunk": "^2.4.2"
   },
   "devDependencies": {
-    "mongodb-data-service": "^22.19.0",
+    "mongodb-data-service": "^22.18.1",
```
Comment (himanshusinghs) [inline packages/compass-sidebar/package.json:73]:
```suggestion
    "mongodb-data-service": "^22.19.0",
```
We can probably keep the most recent one?

Comment (addaleax) [reply packages/compass-sidebar/package.json:73]: Yeah, I don't get it ... why did this script even modify these in the first place 🤷‍♀️

Comment (himanshusinghs) [reply packages/compass-sidebar/package.json]: Yea this happened to me as well when I tried updating electron once with this script. I believe the node-abi package update changes were also automatic, right?

**Review by gribnoysup — APPROVED** [review]

**Review by himanshusinghs — APPROVED** [review]
🚀

**Pull request merged**